lib/truffle/stringio.rb

```diff
@@ -221,17 +221,16 @@ def write(str)
       string.byte_append str
       d.pos = string.bytesize
     elsif pos > string.bytesize
-      m = Truffle::Mirror.reflect string
-      m.splice string.bytesize, 0, "\000" * (pos - string.bytesize)
+      replacement = "\000" * (pos - string.bytesize)
+      Truffle.invoke_primitive(:string_splice, string, replacement, string.bytesize, 0, string.encoding)
       string.byte_append str
       d.pos = string.bytesize
     else
       stop = string.bytesize - pos
       if str.bytesize < stop
         stop = str.bytesize
       end
-      m = Truffle::Mirror.reflect string
-      m.splice pos, stop, str
+      Truffle.invoke_primitive(:string_splice, string, str, pos, stop, str.encoding)
```

Comment:
Is it intentional to take the encoding of the replacement here and not `string.encoding`?
Could the primitive just get the encoding out of the source String's Rope?